app/controllers/authors_controller.rb

```diff
@@ -0,0 +1,22 @@
+class AuthorsController < ActionController::Base
+  def new
+    @author = Author.new
+  end
+
+  def create
+    l_params = {
+      first_name: params[:first_name],
+      last_name: params[:last_name],
+      email: params[:email]
+    }
+
+    @author = Author.new(l_params)
+
+    if @author.valid?
+      RequestJob.perform_later l_params if @author.valid?
+      render :new, status: :ok
+    else
+      render :new, status: 422
```

Comment:
What happens at user side when they enter wrong input? How useful is it to rerender the web form and pass some "unknown" HTTP status code?

Reply:
I didn't know that we actually care for the view as it is not even mentioned to add a form in task requirements. But I used `puts` to output the result to make sure the app works correctly.

422 (Unprocessable Entity) is the status code that 3rd party web service returns when wrong data is passed in the body of the request. As long as I didn't test how the actual API works I returned status code 422 through the controller to make it easy to test.

Reply:
> What happens at user side when they enter wrong input? How useful is it to rerender the web form and pass some "unknown" HTTP status code?

I didn't know that we actually care for the view as it is not even mentioned to add a form in task requirements. But I used puts to output the result to make sure the app works correctly.

422 (Unprocessable Entity) is the status code that 3rd party web service returns when wrong data is passed in the body of the request. As long as I didn't test how the actual API works I returned status code 422 through the controller to make it easy to test.

Reply:
This is clear but as we have the web form its controller should not look like controller of web services. The tests are important but not more important than intended functionality. What's the purpose of form fields validations if you don't communicate the errors back to user?

Reply:
I will refactor the view and controller.